doc/user/content/ingest-data/spill-to-disk.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Spill to Disk"
+description: "How to run and manage Upsert and Debezium sources with spill to disk enabled"
+menu:
+  main:
+    parent: "kafka"
+    name: "Spill to Disk"
+    weight: 4
+---
+
+{{< private-preview />}}
+
+{{< warning >}}
+Pricing for this feature is likely to change in the future.
+{{< /warning >}}
+
+This is a guide on how and why to run Upsert and Debezium Kafka sources with spill to disk.
+
+## Overview
+
+Kafka sources that use `ENVELOPE UPSERT` or `ENVELOPE DEBEZIUM` require the entire _working set_ to be stored locally within its scheduled cluster to produce retractions when keys are updated.
+The working set is sum total of _current_ values for every key.
+
+By default, Materialize uses an in-memory map to store the working set. However, sources scheduled to run on a cluster with an attached disk store this state on that attached disk. Disk-enabled clusters can reduce source memory usage, allowing users to leverage smaller cluster sizes.
```

Comment:
One nit:

while we want to use `spill-to-disk` for the feature, when we actually are talking about replicas, we use "disk-attached replicas", and we should link to this section: https://materialize.com/docs/sql/create-cluster-replica/#disk-attached-replicas here!